gemrb/core/Scriptable/Actor.cpp

```diff
@@ -9546,21 +9566,21 @@ void Actor::SetOverlay(unsigned int overlay)
 //returns true if spell state is already set or illegal
 bool Actor::SetSpellState(unsigned int spellstate)
 {
-	if (spellstate >= SS_MAX) return true;
-	unsigned int pos = IE_SPLSTATE_ID1+(spellstate>>5);
+	if (spellstate >= SpellStatesSize << 5) return true;
+	unsigned int pos = spellstate>>5;
 	unsigned int bit = 1<<(spellstate&31);
-	if (Modified[pos]&bit) return true;
-	Modified[pos]|=bit;
+	if (spellStates[pos]&bit) return true;
+	spellStates[pos]|=bit;
 	return false;
 }
 
 //returns true if spell state is already set
 bool Actor::HasSpellState(unsigned int spellstate) const
 {
-	if (spellstate >= SS_MAX) return false;
```

Comment:
SS_MAX is now an unused macro — please remove it too. No other comments, looks good!

Reply:
Dang, I knew I forgot something!